### ISSUE TITLE
Floating point exception in group_norm

### ISSUE THREAD

**Issue by neochristou** [pytorch/pytorch#77897]
### 🐛 Describe the bug

Floating point exception in `group_norm`.

### Example to reproduce

```python
import torch

input = torch.full((1, 1, 1, 1, 1, 1, 1, 1, 1, 1,), 1, dtype=torch.int64, requires_grad=False)
num_groups = 0
weight = torch.full((1, 1, 1, 1, 1, 1, 1, 1, 1, 1,), 1, dtype=torch.int64, requires_grad=False)
bias = torch.full((1, 1, 1, 1, 1, 1, 1, 1, 1, 1,), 1, dtype=torch.int64, requires_grad=False)
eps = 0
cudnn_enabled = True
torch.group_norm(input, num_groups, weight, bias, eps, cudnn_enabled)
```

### Result
```floating point exception```

### Expected behavior
Graceful termination or a RuntimeError to be thrown.

### Note
This bug was discovered using IvySyn, a fuzz testing tool which is currently being developed at Secure Systems Labs at Brown University.

### Versions

Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: Debian GNU/Linux 11 (bullseye) (x86_64)
GCC version: (Debian 10.2.1-6) 10.2.1 20210110
Clang version: 11.0.1-2
CMake version: version 3.18.4
Libc version: glibc-2.31

Python version: 3.9.7 (default, Sep 16 2021, 13:09:58)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.10.0-11-amd64-x86_64-with-glibc2.31
Is CUDA available: N/A
CUDA runtime version: Could not collect
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A

Versions of relevant libraries:
[pip3] numpy==1.22.3
[pip3] torch==1.11.0
[pip3] torchaudio==0.11.0
[pip3] torchvision==0.12.0
[conda] blas                      1.0                         mkl
[conda] cpuonly                   2.0                           0    pytorch
[conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
[conda] mkl                       2021.4.0           h06a4308_640
[conda] mkl-service               2.4.0            py39h7f8727e_0
[conda] mkl_fft                   1.3.1            py39hd3c417c_0
[conda] mkl_random                1.2.2            py39h51133e4_0
[conda] numpy                     1.21.2           py39h20f2e39_0
[conda] numpy-base                1.21.2           py39h79a1101_0
[conda] pytorch                   1.11.0              py3.9_cpu_0    pytorch
[conda] pytorch-mutex             1.0                         cpu    pytorch
[conda] torchaudio                0.11.0                 py39_cpu    pytorch
[conda] torchvision               0.12.0                 py39_cpu    pytorch

**Comment by ngimel** [comment]
DUplicate of #73188

**Comment by ngimel** [comment]
We'll accept a PR fixing this

**Comment by nishantb21** [comment]
Is this issue still open? Can I take it up?

**Comment by ngimel** [comment]
#73188 is open, the PR linked didn't add the test so we didn't land it. It would be great if you can take over.

**Comment by nishantb21** [comment]
Hi. I did some research about this issue and it appears to be fixed in this PR [#75270 ](https://github.com/pytorch/pytorch/pull/75270/). PR [#73577](https://github.com/pytorch/pytorch/pull/73577) wasn't accepted because of missing test cases although the accepted PR also didn't have test cases. So now executing the above portion of code gives the following error instead of the initial floating point exception it was giving before:

`
RuntimeError: Expected num groups to be greater than 0, got 0
`

Does this still warrant a test case? If so, then I have added a test case in test/test_nn.py but there is an issue with that as well. As you can see the current version of the code raises a RuntimeError when in reality it seems like it should be a ValueError instead? Sorry for asking so many questions, I am new to contributing to open source code. Thank you for your help!

**Comment by ngimel** [comment]
Yes, it definitely warrants a test case (cc @bdhirsh, we shouldn't accept PRs without test cases except when tests are unreasonably hard), best way to add tests like this is to add error_inputs to OpInfo entry for corresponding op in common_methods_invocations.py. Also, feel free to change error type to ValueError. Thanks in advance for you help!
